bin/node-esm

```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env sh
+exec node -r esm $*
```

Comment:
This does not work with global installs at the moment, because the -r esm does not resolve to the global esm install.

Not sure what the best way of fixing that is. At first thought, two major options are apparent to me:
a. I have deployed a wide wide variety of ridiculous crazy shell scripting hacks in the past to detect where the source script (node-esm) is located but every single one has left me feeling not-great. They could be deployed here.
a. Alternatively, this script could itself be converted to be a small node script that forks/spawns/execs the actual target in a similar manner: those scripts would have use of __dirname to help them, which is cleaner. I feel like there's a huge judgement call to be made, a fractal design space this general approach opens up.
a.1. Load esm & try to start the target in the node-esm script directly?
a.1. Try to use something on child_process to launch a new process? I generally don't like this approach because node-esm- instead of a "replacement" for the node executable- becomes a launcher that starts the actual desired work, in a different, new process, & that introduces significant complexity for anyone trying to script/manage/operate tools that make use of any such node-esm.

Both of these paths started to get complicated fast, so I left this PR at a broken-but-conceptually clean & understandable state.